src/ThreeJsPanel.ts

```diff
@@ -713,41 +717,53 @@ export class ThreeJsPanel {
 
     // RENDERING
     // Step 1: Render meshes, e.g. isosurfaces, separately to a render target. (Meshes are all on
-    //   layer 1.) This is necessary to access the depth buffer.
+    // layer 1.) This is necessary to access the depth buffer.
     this.camera.layers.set(MESH_LAYER);
     this.renderer.setRenderTarget(this.meshRenderTarget);
     this.renderer.render(this.scene, this.camera);
 
-    // Step 2: Render the mesh render target out to the screen.
+    // Step 2. Render any passes that have to happen after the meshes are
+    // rendered but before volume rendering (e.g. pick buffer).
+    this.postMeshRenderFuncs.forEach((func) => {
+      func(this.renderer, this.camera, this.meshRenderTarget.depthTexture);
+    });
+
+    // Step 3: Render meshes that do not interact with the pick buffer. This
+    // must happen after the pick buffer is rendered so picking isn't occluded
+    // by them, but before the volume renders so that volumes can still depth
+    // test against the lines.
+    this.renderer.autoClear = false;
+    this.camera.layers.set(MESH_NO_PICK_OCCLUSION_LAYER);
+    this.renderer.setRenderTarget(this.meshRenderTarget);
+    this.renderer.render(this.scene, this.camera);
```

Comment:
Dan: Add an `if` check and skip this step if there are no lines

Reply:
@toloudis tinkered around with this and there wasn't really an elegant way to do this, since `ThreeJsPanel` doesn't have direct access to `VolumeDrawable`. I could have `View3D` be responsible for setting/removing a flag in `ThreeJsPanel` (`hasNoPickMeshes = true/false`), do you think it's worth it for the optimization?